### DISCUSSION TITLE
HTTP status code: 429 (Too Many Requests)

### DISCUSSION THREAD

**Discussion by absoluteweb** [tectalichq/public-openai-client-php#28] [Q&A] [answered]
Hi,
From my php script I often get a fatal error about too many requests. Is this purely related to OpenAI and I need to time the calls, or can it come from the tectalic client?
Fatal error: Uncaught Tectalic\OpenAi\ClientException: Unsuccessful response. HTTP status code: 429 (Too Many Requests). in /home/xxxx/vendor/tectalic/openai/src/Handlers/ChatCompletions.php:150 Stack trace: #0 /home/xxxxx.php(633): Tectalic\OpenAi\Handlers\ChatCompletions->toModel() 
Nicolas.

**Comment by absoluteweb** [comment]
Isn't there a way to avoid making a fatal error, to delay and try again?

**Comment by the-csaba** [comment, marked as answer]
Hi Nicolas,

The "Too Many Requests" error comes from the OpenAI API, not the Tectalic client. It means your application is hitting the rate limits set by OpenAI. 

One way to handle this is by implementing a retry strategy. For example, you could catch the `Tectalic\OpenAi\ClientException` when the status code is 429, then pause for a few seconds and retry the request.

Here is an example of how to do it:

```php
try {
    // your code here
} catch (Tectalic\OpenAi\ClientException $e) {
    if ($e->getCode() == 429) {
        // sleep for a few seconds
        sleep(5);
        // retry the request
    } else {
        throw $e; // rethrow the exception if it's not a 429 error
    }
}
```

This simple approach may not be suitable for every application, especially for those making many requests. You might want to look into more sophisticated rate limiting or backoff strategies for those cases.

I hope this helps!

Csaba

**Reply by absoluteweb** [reply]
Hi,
Thank you very much, I will implement this in my script.
Nicolas.
